erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -30,6 +30,13 @@ frappe.ui.form.on('Payment Entry', {
 				}
 			}
 		});
+		frm.set_query("bank_account", function() {
+			return{
+				"filters": {
+					"party": frm.doc.party,
```

Comment:
One bank account can be linked to multiple parties
So you meant companies account right? In that case user has to make customer against the company name and not against the individual person

Reply:
Added default bank account in the Customer/Supplier

Reply:
Remove the Party field from the Bank Account. This does not make much sense to me, in most cases company uses same bank account to pay / receives payment from multiple parties.

Reply:
In case bank account is not mapped to party in Bank Account doctype, then while linking with Supplier, we would not be able to filter out the bank accounts for the Supplier. Also, multiple bank accounts for a Supplier is possible as bigger companies may have separate bank accounts branch wise.

Reply:
@PawanMeh We have realised the actual use case of party field in bank account. It is used when bank account is actually belongs to that party and we maintain it for Payment Request. We are not removing that field. But we will remove the filtering part, because in Payment Entry it is only involved to company's bank account.